libs/application-generic/src/usecases/upsert-preferences/upsert-preferences.usecase.ts

```diff
@@ -112,7 +124,8 @@ export class UpsertPreferences {
 
     if (command.preferences === null) {
       if (!foundPreference) {
-        throw new BadRequestException('Preference not found');
+        // Gracefully return if the preference is not found
+        return;
```

Comment:
Needed to allow for safe upserts when provided with `null` in the case of a User-Workflow preference set, ensuring the operation can be idempotent.